src/resources/ResourceSnapshots/ResourceSnapshots.ts

```diff
@@ -45,19 +45,52 @@ export default class ResourceSnapshots extends Resource {
         return await fetch(url, {method: 'get'});
     }
 
-    createFromFile(file: File, options: CreateFromFileOptions) {
-        const computedOptions = {developerNotes: options.developerNotes, snapshotFileType: undefined};
+    /**
+     * Creates a snapshot from a file containing the configuration.
+     *
+     * @param {File} file The file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(file: File, options: CreateFromFileOptions): Promise<ResourceSnapshotsModel>;
+
+    /**
+     * Creates a snapshot from a file buffer containing the configuration.
+     *
+     * @param {Buffer} file The file containing the configuration.
+     * @param {ResourceSnapshotSupportedFileTypes} fileType The type of the file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(
+        file: Buffer,
+        fileType: ResourceSnapshotSupportedFileTypes,
+        options: CreateFromFileOptions
+    ): Promise<ResourceSnapshotsModel>;
+
+    createFromFile(
+        file: File | Buffer,
+        typeOrOptions: ResourceSnapshotSupportedFileTypes | CreateFromFileOptions,
+        options?: CreateFromFileOptions
+    ) {
+        let value: File | string;
+        let computedOptions = {developerNotes: undefined, snapshotFileType: undefined};
 
-        if (file.type === 'application/zip') {
-            computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.ZIP;
-        } else if (file.type === 'application/json') {
-            computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.JSON;
+        if (Buffer.isBuffer(file)) {
+            value = file.toString();
+            computedOptions = {developerNotes: options.developerNotes, snapshotFileType: typeOrOptions};
         } else {
-            throw new Error('The uploaded file must be either a ZIP or a JSON file.');
+            const type = file.type;
```

Comment:
There should be an assignation of the `value` variable in the `else` bloc, otherwise when the file param is not a buffer, we will get `form.append('file', undefined)` on line 93.

I don't understand how the existing UTs didn't pick that up 😬 🤔

Reply:
Oups... Good catch.
The UTs are not testing what is being passed to the formData. So, I added an additional UT to ensure the file content is appended to the form data (when passing a buffer).
However, I can hardly do the same thing when passing a File object since `new File(...)` will return a `mockedFile`, which will lead to a `form.append('file', undefined)`.